packages/transports/aws-cloudwatch-logs/src/strategies/worker-queue/worker-queue.strategy.ts

```diff
@@ -1,53 +1,63 @@
 import type { Worker } from "node:worker_threads";
-import type { InputLogEvent } from "@aws-sdk/client-cloudwatch-logs";
+import type { SendEventParams } from "../../common.types.js";
 import { addExitHook } from "../../vendor/exit-hook/index.js";
-import type { CloudWatchLogsStrategy, CloudWatchLogsStrategyOptions, ICloudWatchLogsStrategy } from "../common.js";
+import { BaseStrategy } from "../base.strategy.js";
+import LogWorker from "./worker.js?thread";
 import type {
   CloudWatchLogsWorkerQueueOptions,
   WorkerDataOptions,
   WorkerError,
   WorkerEventMessage,
   WorkerResponse,
   WorkerStopMessage,
-} from "./common.js";
-import LogWorker from "./worker.js?thread";
+} from "./worker-queue.types.js";
 
 const MAX_WAIT_TIME = 30000;
 
 // Uses a worker thread to send logs
-class WorkerQueueStrategy implements ICloudWatchLogsStrategy {
-  #options: CloudWatchLogsStrategyOptions;
+export class WorkerQueueStrategy extends BaseStrategy {
   #queueOptions?: CloudWatchLogsWorkerQueueOptions;
-  #errorHandler?: CloudWatchLogsStrategyOptions["onError"];
   #msgErrorHandler: (error: WorkerError) => void;
   #worker?: Worker;
+  #createIfNotExists: boolean;
+
+  constructor(queueOptions?: CloudWatchLogsWorkerQueueOptions) {
+    if (queueOptions?.batchSize !== undefined) {
+      if (queueOptions.batchSize < 1 || queueOptions.batchSize > 10000) {
+        throw new Error("Batch size must be between 1 and 10000");
+      }
+    }
+
+    if (typeof queueOptions?.delay === "number" && queueOptions.delay < 1) {
+      throw new Error("The specified delay is must be bigger than 0");
+    }
+
+    super();
 
-  constructor(options: CloudWatchLogsStrategyOptions, queueOptions?: CloudWatchLogsWorkerQueueOptions) {
-    const { onError, ...rest } = options;
-    this.#options = rest;
     this.#queueOptions = queueOptions;
-    this.#errorHandler = onError;
+    this.#createIfNotExists = queueOptions?.createIfNotExists ?? false;
     this.#msgErrorHandler = (msg: WorkerResponse) => {
       if (msg.type === "error") {
-        onError?.(msg.error);
+        this.onError?.(msg.error);
       }
     };
 
     addExitHook(this.cleanup.bind(this), { wait: MAX_WAIT_TIME });
   }
 
-  public sendEvent(event: InputLogEvent, logGroupName: string, logStreamName: string): void {
+  public sendEvent({ event, logGroupName, logStreamName }: SendEventParams): void {
```

Comment:
changed the signature of sendEvent to be a sole object so it'll be easier to expand upon in the future if we need to add more parameters